osm_tile_data_extract/import_extracts.py

```diff
@@ -0,0 +1,120 @@
+        subprocess.run(['su', 'postgres', '-c', 'echo "CREATE USER tile_data PASSWORD \'tile_data\';" | psql'],
+                       check=True, stdout=subprocess.DEVNULL)
+        subprocess.run(
+            ['su', 'postgres', '-c', f'echo "CREATE DATABASE {self.db_dbname} OWNER tile_data;" | psql'],
+            check=True, stdout=subprocess.DEVNULL)
+        subprocess.run(['su', 'postgres', '-c', f'echo "CREATE EXTENSION postgis;" | psql -d {self.db_dbname}'],
+                       check=True, stdout=subprocess.DEVNULL)
+        subprocess.run(['su', 'postgres', '-c', f'echo "CREATE EXTENSION hstore;" | psql -d {self.db_dbname}'],
+                       check=True, stdout=subprocess.DEVNULL)
+
+    def _download_shapefiles(self):
+        print_stage('Downloading shapefiles')
+        bootstrap_dir = os.path.join(self.working_dir, 'vector-bootstrap')
+        os.makedirs(bootstrap_dir, exist_ok=True)
+        subprocess.run(['python', './bootstrap.py'], cwd='/app/src/vector-datasource/data', check=True)
+        subprocess.run(['make', '-f', '/app/src/vector-datasource/data/Makefile-import-data'],
+                       check=True, cwd=bootstrap_dir)
+
+    def _import_into_postgres(self):
+        print_stage("Importing data with osm2pgsql")
+        subprocess.run([
+            'su', 'postgres', '-c',
+            f'osm2pgsql --slim --hstore-all -C 3000 '
+            f'-S /app/src/vector-datasource/osm2pgsql.style '
+            f'-d {self.db_dbname} '
+            f'-P {self.db_port} '
+            f'-U postgres '
+            f'--number-processes {max(1, os.cpu_count() - 2)} '
+            f'{os.path.join(self.working_dir, self.pbf_file_name)}'
+        ], check=True)
+
+    def _post_import(self):
+        print_stage('Importing shapefiles')
+        subprocess.run(['su', 'postgres', '-c', f'./import-shapefiles.sh | psql -d {self.db_dbname}'],
+                       check=True, cwd=os.path.join(self.working_dir, 'vector-bootstrap'))
+
+        print_stage('Processing imported data')
+        subprocess.run(['su', 'postgres', '-c', f'./perform-sql-updates.sh -d {self.db_dbname}'],
+                       check=True, cwd='/app/src/vector-datasource/data')
+
+    def _dump_postgres_db(self):
+        print_stage('Dumping PostgreSQL database')
+        file_path = os.path.join(self.working_dir, 'db.pg_dump')
+        with open(file_path, 'wb') as f:
+            subprocess.run(['su', 'postgres', '-c',
+                            f'pg_dump -p {self.db_port} -d {self.db_dbname} --format custom'],
+                           check=True, cwd=self.out_dir, stdout=f)
+        subprocess.run(['chown', '0:0', file_path])
+
+    def _upload_dump(self):
+        print_stage('Uploading PostgreSQL dump to tileserver-mapping')
+        file_path = os.path.join(self.working_dir, 'db.pg_dump')
+        self.api.upload_sql_dump(self.tile, file_path)
+        subprocess.run(['rsync', file_path, self.out_dir], check=True)
```

Comment:
if it is uploaded i would not put it into local storage, it would only waste space